### PR TITLE
Handle closing modal by outside click and escape key

### PR DESCRIPTION
**Pull request by CPatchane**
In order to save @y-lohse for the next demos ;)

### CODE REVIEW

File: react/Modal/index.jsx [hunk 1]
```diff
@@ -49,9 +49,27 @@ const ModalButtons = ({ secondaryText, secondaryAction, secondaryType, primaryTe
...
+  document.addEventListener('keyup', handleKeyUp)
+
+  const handleOutsideClick = (event) => {
+    if (event.target.className === overlayClassName) props.secondaryAction()
+  }
+
   return (
     <div className={styles['coz-modal-container']}>
-      <div className={styles['coz-overlay']}>
+      <div onClick={handleOutsideClick} className={styles['coz-overlay']}>
```
Comment (y-lohse) [inline react/Modal/index.jsx]: You might use `overlayClassName` here just in case, but whatever :-)

File: react/Modal/index.jsx [hunk 2]
```diff
@@ -49,14 +49,27 @@ const ModalButtons = ({ secondaryText, secondaryAction, secondaryType, primaryTe
 
 const Modal = (props) => {
   const children = props.children
+  const overlayClassName = styles['coz-overlay']
 
-  const onOutsideClick = (event) => {
-    if (event.target.className === styles['coz-overlay']) props.secondaryAction()
+  const handleKeyUp = (event) => {
+    if (document.querySelector(`.${overlayClassName}`)) {
+      if (event.keyCode === 27) {
```
Comment (gregorylegarec) [inline react/Modal/index.jsx]: I suggest to use a constant for the keyCode, to clarify code.
```js
if (event.keyCode === KEY_ESC)
```

File: react/Modal/index.jsx (hunk 1 above)
Comment (CPatchane) [inline react/Modal/index.jsx]: You're right, I fix that 👍

**Comment by CPatchane** [comment]
Thanks @ptbrowne I was hesitated about making a whole react component, but it seems cleaner when seeing your proposal. It's a component now 👍

**Pull request merged**